internal/services/autopi/hardware_template_service.go

```diff
@@ -151,6 +152,24 @@ func (a *hardwareTemplateService) ApplyHardwareTemplate(ctx context.Context, req
 		if err != nil {
 			return nil, errors.Wrap(err, "failed to unassociate template")
 		}
+
+		vehicleLoggers := new([]services.AutoPiVehicleLogger)
+		err = a.ap.GetVehicleLoggers(autoPi.Vehicle, vehicleLoggers)
+		if err != nil {
+			return nil, errors.Wrap(err, fmt.Sprintf("failed to get vehicle loggers for vehicleID %d", autoPi.Vehicle.ID))
+		} else {
+			for _, logger := range *vehicleLoggers {
+				if logger.Parent == 0 {
```

Comment:
what does logger.Parent == 0 mean? When looking at loggers returned for my car that were supposed to be removed, they had a parent > 0.

Reply:
saw your comment in the PR description, that parent = 0 means not associated by template - is this something AutoPi said?

Reply:
leave a comment on line 162 explaining why the check for parent == 0. This is usually a good use case for comments in code when there is some otherwise unknown magic required logic.

Reply:
when not associated by template, parent is actually NULL, but since integers cannot be NULL in golang, when converting from the JSON response, the expectation is that "parent" field would populate as value 0. 
I'll update the commit, no problem